javascript-review.js

```diff
@@ -1,92 +1,190 @@
 
-// 1.
-
+// 1. 
 // Create an object that has four properties. One property should be set so that the value is a number, another property set to a string, the third to an array, and the fourth to a function. This function should simply console.log the value of the first property.
+// let familyObj = {
+//         numMembers: 4,
+//         familyName: 'Nikolic',
+//         familyMembers: ['Jovan', 'Alex', 'Vicky', 'David'],
+//         printMembers: function(){
+//             console.log(`The number of people in my family are ${this.numMembers}`);
+//         }
+// }
 
-// 2.
+// familyObj.printMembers()
 
+// 2. 
 // Write a constructor function (to create objects) that takes one argument. It should use that argument to set the value of the first attribute of the object. There should be a second attribute that stores the value ‘red’. 
 
 // Note that a constructor function is the old school syntax that was used in javascript before the sugar class syntax came along. 
 
-// 3.
+// function Animal(name){
+//     // Object.create(Animal); //you only use Object.create() if you want to inherit the methods and attributes from another object
+//     let animal = {};
+//     animal.name = name;
+//     animal.color = 'red';
+//     return animal;
+// }
+
+// let dog = new Animal('dog');
+// console.log(dog);
+
+// let cat = new Animal('cat');
+// console.log(cat);
+
 
+
+// 3. 
 // Can you access variables defined outside of functions within the scope of functions in JS? Show a simple example of this being used? (Define a constant in global scope and use that in a very simple function.)
 
-// 4.
+//Yes you can, you just need to define the variable you want to use outside the functions as a global variable
+// const y = 10;
+
+// const printXAndY = () => {
+//     let x = 20;
+//     console.log(x, y);
+// }
+
+// printXAndY();
 
+
+// 4.
 // What two ways can be used to access the properties of objects in JS? Give examples of both (although assume the objects have been defined).
+// let obj = {
+//     name: 'Jov',
+//     age: '22',
+//     address: 'Yarrambat'
+// }
+
+// console.log(keys(obj));
+
+// you can iterate through and get the objects keys in an array or access them directly if you know them
 
-// 5.
 
 // What is JSON? Be clear but concise.
+// 5. JSON stands for JavaScript Object Notation and is a format for storing data in and object like fashion, where the keys are all strings.
 
-// 6.
 
 // What is a callback?
+// 6. A callback is a call to another function within a function. Callback functions get passed into other functions that utilise them.
```

Comment:
this is right, the key is that a function is being passed as an argument, it doesn't necessarily have to be called